dbms/src/Storages/Page/V3/Universal/S3PageReader.cpp

```diff
@@ -74,12 +74,13 @@ UniversalPageMap S3PageReader::read(const UniversalPageIdAndEntries & page_id_an
     return page_map;
 }
 
-std::pair<UniversalPageMap, UniversalPageMap> S3PageReader::read(const FieldReadInfos & to_read)
+std::pair<UniversalPageMap, UniversalPageMap> S3PageReader::read(FieldReadInfos & to_read)
 {
     UniversalPageMap complete_page_map;
     size_t read_fields_size = 0;
-    for (const auto & read_info : to_read)
+    for (auto & read_info : to_read)
     {
+        std::sort(read_info.fields.begin(), read_info.fields.end());
```

Comment:
When building `std::set<FieldOffsetInsidePage>` later in this method, it will iterate over the `read_info.fields` in order and computes corresponding offsets for each field. Then the `{index, offset}` will be inserted into `std::set<FieldOffsetInsidePage>` in the order of `index`. So `offset` increases as `index` increases, otherwise it will fire `end >= beg` in `Page::getFieldData`